### PR TITLE
feat: CP support Pagination className and style

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | feat: CP support Pagination className and style |
| 🇨🇳 Chinese | -- |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 3d2498e</samp>

This pull request enhances the `ConfigProvider` and `Pagination` components by adding new props and improving the test coverage and documentation. It allows users to customize and localize the `Pagination` component through the `ConfigProvider` component, and passes the `pagination` prop to the `ConfigContext` and the `LocaleProvider`. It also updates the API documentation in both English and Chinese, and adds a new test case for the `ConfigProvider` component.

### <samp>🤖 Generated by Copilot at 3d2498e</samp>

*  Add pagination prop to ConfigProvider component to allow setting common props for Pagination component ([link](https://github.com/ant-design/ant-design/pull/43100/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R117-R118), [link](https://github.com/ant-design/ant-design/pull/43100/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R228), [link](https://github.com/ant-design/ant-design/pull/43100/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R278), [link](https://github.com/ant-design/ant-design/pull/43100/files?diff=unified&w=0#diff-975af6d8b4e359c4e88a7586cf1c91b35989cdadc9f2dc9ca16a376b5f5b0563R68), [link](https://github.com/ant-design/ant-design/pull/43100/files?diff=unified&w=0#diff-b6a7375df96aaccf21f7452705af80f8cd8c7323f0fc2b9dbdc71500b8478775R69))
*  Add className and style props to ConfigConsumerProps interface to pass them from ConfigProvider to Pagination component through context ([link](https://github.com/ant-design/ant-design/pull/43100/files?diff=unified&w=0#diff-f7a3a5082a18fdd0627b75fb4f13fa559a2e5a5a781e381b8ba8c98b13318ccaR64-R65))
*  Use pagination.className and pagination.style from context to apply custom class and style to Pagination component ([link](https://github.com/ant-design/ant-design/pull/43100/files?diff=unified&w=0#diff-6ded05884b813a9ee306b10152a2e379ac9c8c724ade740984c1aed485d14352R114), [link](https://github.com/ant-design/ant-design/pull/43100/files?diff=unified&w=0#diff-6ded05884b813a9ee306b10152a2e379ac9c8c724ade740984c1aed485d14352L117-R126), [link](https://github.com/ant-design/ant-design/pull/43100/files?diff=unified&w=0#diff-6ded05884b813a9ee306b10152a2e379ac9c8c724ade740984c1aed485d14352R42))
*  Add test case to check if ConfigProvider can pass className and style props to Pagination component ([link](https://github.com/ant-design/ant-design/pull/43100/files?diff=unified&w=0#diff-0240afe2faa6807b920ad4c86a45d6724eb47ed4cde840f097b69fdb4d28cdb5R9), [link](https://github.com/ant-design/ant-design/pull/43100/files?diff=unified&w=0#diff-0240afe2faa6807b920ad4c86a45d6724eb47ed4cde840f097b69fdb4d28cdb5R202-R214))

### CODE REVIEW

File: components/config-provider/index.en-US.md [hunk 1]
```diff
@@ -107,6 +107,7 @@ const {
 | form | Set Form common props | { validateMessages?: [ValidateMessages](/components/form/#validatemessages), requiredMark?: boolean \| `optional`, scrollToFirstError?: boolean \| [Options](https://github.com/stipsan/scroll-into-view-if-needed/tree/ece40bd9143f48caf4b99503425ecb16b0ad8249#options) } | - | requiredMark: 4.8.0; colon: 4.18.0; scrollToFirstError: 5.2.0 |
 | image | Set Image common props | { className?: string, style?: React.CSSProperties } | - | 5.7.0 |
 | input | Set Input common props | { autoComplete?: string } | - | 4.2.0 |
+| pagination | Set Pagination common props | { className?: string, style?: React.CSSProperties } | - | 5.7.0 |
```
Comment (MadCcc) [inline components/config-provider/index.en-US.md:110]: showSizeChanger 也加上吧

Comment (li-jia-nan) [reply components/config-provider/index.en-US.md]: done

File: components/config-provider/context.ts [hunk 2]
```diff
@@ -62,6 +62,8 @@ export interface ConfigConsumerProps {
   };
   pagination?: {
     showSizeChanger?: boolean;
+    className?: string;
+    style?: React.CSSProperties;
```
Comment (MadCcc) [inline components/config-provider/context.ts:66]: 有个 CommonConfig，& 一下？

**Review by MadCcc — APPROVED** [review]

File: components/config-provider/context.ts (hunk 2 above)
Comment (li-jia-nan) [reply components/config-provider/context.ts]: done

**Pull request merged**